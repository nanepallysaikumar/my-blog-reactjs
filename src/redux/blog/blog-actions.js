import {
    BLOG_UPDATE_LIST,
    BLOG_ADD_ITEM,
    BLOG_EDIT_ITEM,
    BLOG_REMOVE_ITEM
} from './blog-constants';
import { createApiActions } from './blog-api';

export function blogList(items) {
    return {
        type: BLOG_UPDATE_LIST,
        payload: items
    };
}

export function addBlog(blog) {
    return {
        type: BLOG_ADD_ITEM,
        payload: blog
    };
}

export function editBlog(blog) {
    return {
        type: BLOG_EDIT_ITEM,
        payload: blog
    };
}

export function removeBlog(blog) {
    return {
        type: BLOG_REMOVE_ITEM,
        payload: blog
    };
}

export const blogApiActions = createApiActions(fetch);
