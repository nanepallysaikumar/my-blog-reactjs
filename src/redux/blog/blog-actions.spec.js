import { blogList, addBlog, editBlog, removeBlog } from './blog-actions';

describe('blog-actions', () => {
    const blog = {
        heading: 'test heading',
        body: 'test body'
    };

    describe('blogList', () => {
        it('should generate valid action object', () => {
            expect(blogList([blog])).toMatchSnapshot('blogList');
        });
    });
    describe('addBlog', () => {
        it('should generate valid action object', () => {
            expect(addBlog(blog)).toMatchSnapshot('addBlog');
        });
    });

    describe('editBlog', () => {
        it('should generate valid action object', () => {
            expect(editBlog(blog)).toMatchSnapshot('editBlog');
        });
    });

    describe('removeBlog', () => {
        it('should generate valid action object', () => {
            expect(removeBlog(blog)).toMatchSnapshot('removeBlog');
        });
    });
});
