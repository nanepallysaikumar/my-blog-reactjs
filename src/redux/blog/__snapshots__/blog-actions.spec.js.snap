// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog-actions addBlog should generate valid action object: addBlog 1`] = `
Object {
  "payload": Object {
    "body": "test body",
    "heading": "test heading",
  },
  "type": "BLOG_ADD_ITEM",
}
`;

exports[`blog-actions blogList should generate valid action object: blogList 1`] = `
Object {
  "payload": Array [
    Object {
      "body": "test body",
      "heading": "test heading",
    },
  ],
  "type": "BLOG_UPDATE_LIST",
}
`;

exports[`blog-actions editBlog should generate valid action object: editBlog 1`] = `
Object {
  "payload": Object {
    "body": "test body",
    "heading": "test heading",
  },
  "type": "BLOG_EDIT_ITEM",
}
`;

exports[`blog-actions removeBlog should generate valid action object: removeBlog 1`] = `
Object {
  "payload": Object {
    "body": "test body",
    "heading": "test heading",
  },
  "type": "BLOG_REMOVE_ITEM",
}
`;
